### DISCUSSION TITLE
How to develop on an application and on a Nuxt module at the same time?

### DISCUSSION THREAD

**Discussion by Ambrise** [nuxt/nuxt#25622] [Questions]
Hello everyone, 

I would like to know how you manage to work at the same time on a Nuxt module and a final application with Nuxt 3. I understand that to develop a module, you just need to use playground to test your module . But in my case I have a nuxt application which is almost finished but I would like to move certain pieces of code into a Nuxt module. I would like to be able to run my final application locally and which points to my module.

My two projects are in the same folder. Do you use "npm link" for that? 

Thank you in advance and have a very good day. 

Sincerely. 
Axel

**Comment by xjccc** [comment]
You can use folder `modules`
[modules](https://nuxt.com/docs/guide/directory-structure/modules)

or 

```
export default defineNuxtConfig({
  modules: [
    // Load a local module
    './modules/example'
  ]
})

```
[add modules](https://nuxt.com/docs/guide/concepts/modules#add-nuxt-modules)

**Reply by Ambrise** [reply]
Thank you for your reply ! But if the module is a full-fledged Git project, I can't put it in the modules folder, it must be external to my application?
Good day to you

**Reply by shvchuk** [reply]
I used to work on the project where we had our components library as a nuxt module. We used `npm link` to work on both (which is adding new features to components and testing the outcome in the application).

**Reply by xjccc** [reply]
emm.. I think, if your project of nuxt modules at local.

You also could use your local module path?

**Comment by Ambrise** [comment]
I tried like this, but I don't know if it's a good practice ?

app/nuxt.config.ts :
```
modules: [
        '../mymodule/src/module.ts'
    ],
```

![image](https://github.com/nuxt/nuxt/assets/73852515/851b1b77-33a2-45ee-a626-94057ff6c10b)

Thanks a lot for your answers
